src/diff/index.js

```diff
@@ -443,7 +443,11 @@ function diffElementNodes(
 			} else if (i == 'dangerouslySetInnerHTML') {
 				oldHtml = value;
 			} else if (i !== 'key' && !(i in newProps)) {
-				if (i == 'value' && 'defaultValue' in newProps) continue;
+				if (
+					(i == 'value' && 'defaultValue' in newProps) ||
+					(i == 'checked' && 'defaultChecked' in newProps)
+				)
+					continue;
```

Comment:
tiny nit, let's add proper braces here. I'm fine with leaving them out when it's a single line condition like before